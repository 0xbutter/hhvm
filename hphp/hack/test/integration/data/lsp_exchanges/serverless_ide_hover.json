[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{
        "namingTableSavedStatePath":"${naming_table_saved_state_path}"
      },
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":{}
    }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "comment": "hover over function invocation",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 2,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 16}
   }
},
{
  "comment": "hover over whitespace",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 3,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 1}
   }
},
{
  "comment": "hover over a keyword",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 4,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 2, "character": 1}
   }
},
{
  "comment": "hover over a comment",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 5,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 1, "character": 4}
   }
},
{
  "comment": "hover past the end of a line",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 6,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 100}
   }
},
{
  "comment": "hover past the end of a file",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 7,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 300, "character": 0}
   }
},
{
  "comment": "make local, unsaved change to the file",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
    "textDocument": {
      "uri": "${php_file_uri}",
      "version": 2
    },
    "contentChanges": [
      {
        "text": "<?hh  //strict\n// comment\nfunction a_hover(): int {\n  return b_hover();\n}\n\n# Another comment describing b_hover.\nfunction b_hover(): int {\n  return 42;\n}\n"
      }
    ]
  }
},
{
  "comment": "another hover over function invocation",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 8,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 16}
   }
},
{
  "jsonrpc": "2.0",
  "id": 99,
  "method": "shutdown",
  "params": {}
}
]
