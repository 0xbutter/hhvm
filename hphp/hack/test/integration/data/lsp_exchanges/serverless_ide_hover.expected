[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "typeDefinitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "telemetry/event",
    "params": {
      "type": 1,
      "message": "connect failed: No_server_running_should_retry [-32001] [remaining 11 lines truncated for testing...]"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "client/registerCapability",
    "params": {
      "registrations": [
        {
          "id": "did-change-watched-files",
          "method": "workspace/didChangeWatchedFiles",
          "registerOptions": {
            "watchers": [
              {
                "globPattern": "**",
                "kind": 7
              }
            ]
          }
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": {
      "contents": [
        {
          "language": "hack",
          "value": "int"
        },
        "A comment describing b_hover."
      ],
      "range": {
        "start": {
          "line": 3,
          "character": 9
        },
        "end": {
          "line": 3,
          "character": 16
        }
      }
    },
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": null,
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "result": null,
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "result": null,
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "result": null,
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "result": null,
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 8,
    "result": {
      "contents": [
        {
          "language": "hack",
          "value": "int"
        },
        "Another comment describing b_hover."
      ],
      "range": {
        "start": {
          "line": 3,
          "character": 9
        },
        "end": {
          "line": 3,
          "character": 16
        }
      }
    },
    "powered_by": "serverless_ide"
  },
  {
    "jsonrpc": "2.0",
    "id": 99,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "method": "telemetry/event",
    "params": {
      "type": 1,
      "message": "hh_server stopped [-32800] [remaining 5 lines truncated for testing...]"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "telemetry/event",
    "params": {
      "type": 1,
      "message": "hh_server stopped [-32800] [remaining 5 lines truncated for testing...]"
    }
  }
]
